web/frontend/src/components/slam-2d-render.vue

```diff
@@ -78,40 +78,44 @@ const disposeScene = () => {
 };
 
 const updateCloud = (pointcloud: Uint8Array) => {
-  disposeScene();
-
-  const viewHeight = 1;
-  const viewWidth = viewHeight * 2;
-
-  const points = loader.parse(pointcloud.buffer, '');
-  points.geometry.computeBoundingSphere();
-
-  const { radius = 1, center = { x: 0, z: 0 } } = points.geometry.boundingSphere ?? {};
-  camera.position.set(center.x, 100, center.z);
-  camera.lookAt(center.x, 0, center.z);
-
-  const aspect = canvas.clientHeight / canvas.clientWidth;
-  camera.zoom = aspect > 1
-    ? viewHeight / (radius * 2)
-    : camera.zoom = viewWidth / (radius * 2);
-
-  camera.updateProjectionMatrix();
-
-  controls.target.set(center.x, 0, center.z);
-  controls.maxZoom = radius * 2;
-
-  const intersectionPlane = new THREE.Mesh(
-    new THREE.PlaneGeometry(radius * 2, radius * 2, 1, 1).rotateX(-Math.PI / 2),
-    new MeshDiscardMaterial()
-  );
-  intersectionPlane.name = 'Intersection Plane';
-  intersectionPlane.position.y = -1;
-  intersectionPlane.position.set(center.x, 0, center.z);
-  raycaster.objects = [intersectionPlane];
-
-  scene.add(points);
-  scene.add(marker);
-  scene.add(intersectionPlane);
+  try {
```

Comment:
This just moves this work into a try catch so that we can log the error if there is a failure, rather than throwing in vue. 

This was recommended by eric.

Reply:
Can we move only the parts that can possibly throw here? There's a lot here that'll never throw and I'd rather be explicit about where we're listening for thrown errors.

Reply:
Thanks @edaniels! I've now moved the try catch to just the watch functions.